Helper/Data.php

```diff
@@ -1167,16 +1168,24 @@ public function getMagentoDetails()
         ];
     }
 
-    public function buildRequestHeaders()
+    public function buildRequestHeaders($payment = null)
     {
         $magentoDetails = $this->getMagentoDetails();
-        return [
+        $headers = [
             'external-platform-name' => $magentoDetails['name'],
             'external-platform-version' => $magentoDetails['version'],
             'external-platform-edition' => $magentoDetails['edition'],
             'merchant-application-name' => $this->getModuleName(),
             'merchant-application-version' => $this->getModuleVersion()
         ];
+
+        if(isset($payment)){
+            $headers[HeaderDataBuilder::FRONTENDTYPE] =
+                $payment->getAdditionalInformation(HeaderDataBuilder::FRONTENDTYPE) ??
+                    '';
+        }
```

Comment:
```suggestion
        if(isset($payment) && !is_null($payment->getAdditionalInformation(HeaderDataBuilder::FRONTENDTYPE))){
            $headers[HeaderDataBuilder::FRONTENDTYPE] = $payment->getAdditionalInformation(HeaderDataBuilder::FRONTENDTYPE);
        }
```

Do we really need to send empty value for frontend type if we don't know the type?